ui/page/wallet_page.go

```diff
@@ -410,7 +425,11 @@ func (pg *WalletPage) layoutOptionsMenu(gtx layout.Context, optionsMenuIndex int
 
 func (pg *WalletPage) walletSection(gtx layout.Context) layout.Dimensions {
 	return pg.walletsList.Layout(gtx, len(pg.listItems), func(gtx C, i int) D {
+
+		pg.listLock.Lock()
 		listItem := pg.listItems[i]
+		pg.listLock.Unlock()
+
```

Comment:
Put a local copy outside the loop to avoid access the mutex in every loop run.
```suggestion
func (pg *WalletPage) walletSection(gtx layout.Context) layout.Dimensions {
	pg.listLock.Lock()
	listItems := pg.listItems
	pg.listLock.Unlock()
	return pg.walletsList.Layout(gtx, len(listItems), func(gtx C, i int) D {

		listItem := pg.listItems[i]

```